arangod/Aql/QueryCursor.cpp

```diff
@@ -380,3 +392,38 @@ Result QueryStreamCursor::writeResult(VPackBuilder& builder, ExecutionState stat
 std::shared_ptr<transaction::Context> QueryStreamCursor::context() const {
   return _query->trx()->transactionContext();
 }
+
+ExecutionState QueryStreamCursor::prepareDump(std::function<void()> const& continueHandler) {
```

Comment:
the continueHandler is unused here.
It needs to be registered on the QUery.
In fact it maybe registered there, as queries will never change RestHandlers for Cursor.